src/main/java/io/cdap/plugin/http/common/http/HttpClient.java

```diff
@@ -132,7 +133,14 @@ public CloseableHttpClient createHttpClient(String pageUriStr) throws IOExceptio
       httpClientBuilder.setProxy(proxyHost);
     }
     httpClientBuilder.setDefaultCredentialsProvider(credentialsProvider);
-
+    ArrayList<Header> clientHeaders = new ArrayList<>();
+    // If OAuth2 is enabled, fetch an access token and add it as an Authorization header.
+    if (Boolean.TRUE.equals(config.getOauth2Enabled())) {
+      AccessToken oauth2AccessToken = OAuthUtil.getAccessToken(HttpClients.createDefault(), config);
```

Comment:
why are we using `HttpClients.createDefault()` here? It will not consider configs added above like request timeout, connection timeout, etc. Please use the above `httpClientBuilder`